Projects/Checkers/Board.cs

```diff
@@ -0,0 +1,171 @@
+			{
+				(int X, int Y) vector = (a.X - b.X, a.Y - b.Y);
+				double distanceSquared = vector.X * vector.X + vector.Y * vector.Y;
+				if (distanceSquared < minDistanceSquared)
+				{
+					minDistanceSquared = distanceSquared;
+					closestRivals = (a, b);
+				}
+			}
+		}
+		return closestRivals;
+	}
+
+	public List<Move> GetPossibleMoves(PieceColor color)
+	{
+		List<Move> moves = new();
+		if (Aggressor is not null)
+		{
+			if (Aggressor.Color != color)
+			{
+				throw new Exception();
+			}
+			moves.AddRange(GetPossibleMoves(Aggressor).Where(move => move.PieceToCapture is not null));
+		}
+		else
+		{
+			foreach (Piece piece in Pieces.Where(piece => piece.Color == color))
+			{
+				moves.AddRange(GetPossibleMoves(piece));
+			}
+		}
+		return moves.Any(move => move.PieceToCapture is not null)
+			? moves.Where(move => move.PieceToCapture is not null).ToList()
+			: moves;
+	}
+
+	public List<Move> GetPossibleMoves(Piece piece)
+	{
+		List<Move> moves = new();
+		ValidateMove(-1, -1);
+		ValidateMove(-1, 1);
+		ValidateMove(1, -1);
+		ValidateMove(1, 1);
+		return moves.Any(move => move.PieceToCapture is not null)
+			? moves.Where(move => move.PieceToCapture is not null).ToList()
+			: moves;
+
+		void ValidateMove(int dx, int dy)
+		{
+			if (!piece.Promoted && piece.Color is Black && dy is -1) return;
+			if (!piece.Promoted && piece.Color is White && dy is 1) return;
+			(int X, int Y) target = (piece.X + dx, piece.Y + dy);
+			if (!Board.IsValidPosition(target.X, target.Y)) return;
+			PieceColor? targetColor = this[target.X, target.Y]?.Color;
+			if (targetColor is null)
+			{
+				if (!Board.IsValidPosition(target.X, target.Y)) return;
+				Move newMove = new(piece, target);
+				moves.Add(newMove);
+			}
+			else if (targetColor != piece.Color)
+			{
+				(int X, int Y) jump = (piece.X + 2 * dx, piece.Y + 2 * dy);
+				if (!Board.IsValidPosition(jump.X, jump.Y)) return;
+				PieceColor? jumpColor = this[jump.X, jump.Y]?.Color;
+				if (jumpColor is not null) return;
+				Move attack = new(piece, jump, this[target.X, target.Y]);
+				moves.Add(attack);
+			}
+		}
+	}
+
+	/// <summary>Returns a <see cref="Move"/> if <paramref name="from"/>-&gt;<paramref name="to"/> is valid or null if not.</summary>
+	public Move? ValidateMove(PieceColor color, (int X, int Y) from, (int X, int Y) to)
+	{
+		Piece? piece = this[from.X, from.Y];
+		if (piece is null)
+		{
+			return null;
+		}
+		foreach (Move move in GetPossibleMoves(color))
+		{
+			if ((move.PieceToMove.X, move.PieceToMove.Y) == from && move.To == to)
+			{
+				return move;
+			}
+		}
+		return null;
+	}
+
+	public bool IsTowards(Move move, Piece piece)
+	{
+		(int Dx, int Dy) a = (move.PieceToMove.X - piece.X, move.PieceToMove.Y - piece.Y);
+		int a_distanceSquared = a.Dx * a.Dx + a.Dy * a.Dy;
+		(int Dx, int Dy) b = (move.To.X - piece.X, move.To.Y - piece.Y);
+		int b_distanceSquared = b.Dx * b.Dx + b.Dy * b.Dy;
```

Comment:
naming is all I would say here, camelCase and avoid underscores in **local variables**

Reply:
in this case "a" and "b" are vectors while "a_distanceSquared" and "b_distanceSquared" are the magnitudes of those vectors squared. I realize the naming convention is a bit strange, but my intent was to show that "a" is related to "a_distanceSquared" and "b" is related to "b_distanceSquared".

Should it be changed? Maybe. I'll resolve this comment for now, but I may change this in the future.